### ISSUE TITLE
setup.py requires numpy to be preinstalled

### ISSUE THREAD

**Issue by betsig** [biocore/scikit-bio#1671]
Having an issue trying to install scikit-bio on a fresh python enviroment (python 3.6.7).

```
$ pip install scikit-bio
Collecting scikit-bio
  Using cached https://files.pythonhosted.org/packages/2d/ff/3a909ae8c212305846f7e87f86f3902408b55b958eccedf5d4349e76c671/scikit-bio-0.5.5.tar.gz
    ERROR: Command errored out with exit status 1:
     command: /shared/homes/143470/python_envs/borf2/bin/python3 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-edv3xz4m/scikit-bio/setup.py'"'"'; __file__='"'"'/tmp/pip-install-edv3xz4m/scikit-bio/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base pip-egg-info
         cwd: /tmp/pip-install-edv3xz4m/scikit-bio/
    Complete output (5 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-edv3xz4m/scikit-bio/setup.py", line 20, in <module>
        import numpy as np
    ModuleNotFoundError: No module named 'numpy'
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output
```
It appears as though in setup.py, `import numpy as np` (line 20) occurs before installing the required packages. (Under `install_requires`).

I can work around this by installing numpy myself, but shouldn't this be fixed? Is there another way around using `np.get_include()` in the setup file?

**Comment by ElDeveloper** [comment]
@betsig, the last time I checked, there's no easy way to work around this problem. However if things have changed and there's now a way to do this, we would welcome a PR to fix this problem.

**Comment by mwilliamson** [comment]
If I've understood correctly, then [PEP 518](https://www.python.org/dev/peps/pep-0518/), which is [supported in pip](https://pip.pypa.io/en/stable/reference/pip/#pep-517-and-518-support), supports this by the creation of a `pyproject.toml` file.

**Comment by mwilliamson** [comment]
I've suggested a fix here: https://github.com/biocore/scikit-bio/pull/1690

**Comment by wasade** [comment]
Fixed in #1690
